Feature/social

Honeybeei opened the pull request:
## ✅ 풀\_리퀘스트 체크리스트

<!--
하나씩 확인 후 체크박스에 표시해주세요.
-->

- [x] PR 제목: [Feature/Fix/Refactor...] 작업 내용 한 줄 요약 (브랜치 이름) #이슈번호
- [x] commit message 가 적절한지 확인해주세요.
- [x] 적절한 branch 로 요청했는지 확인해주세요.
- [x] Assignees, Label 을 붙여주세요.
- [x] 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer 로 등록해주세요.
- [ ] PR이 승인된 경우 해당 브랜치는 삭제해 주세요!

<br/>

## 🔄 변경 사항

<!-- 해당 pr에서 작업한 내역을 적어주세요. 처음엔 간단하게 요약, list 형식으로 세부사항 작성 -->

Social page 에 들어가는 카드들의 기능을 업그레이드 했습니다. 

- Loader 를 만들어보았습니다. 가져다 쓰셔요
- Social page 에서 dummy data 를 가져올 때 async function 으로 변경하여 로더를 사용 할 수 있게 만들었습니다. 

- 각 버튼들에 tooltip(호버링 할 때 설명 나오는것)을 추가했습니다. 
- 각 버튼들의 기능을 구현했습니다. 이제 백엔드만 가져다 붙이면 됩니다. (DM빼고). 현재는 dummy async function 으로 간단하게 구현만 해놨습니다. 요청시 일정 타임 딜레이 후 전송 실패한 경우로 구현해 놨습니다. 전송 성공을 보고싶으면 코멘트 달려 있는 부분을 true로 바꿔보세요!
- shadcn컴포넌트중 toast의 css를 사알짝 변경했습니다. 앞으로 덮어쓰기 주의!!!

<br/>

## 📎 변경한 이유

- 이것이 enhancement 이니까

<br/>

<!-- 관련되어있는 Issue Number 를 작성하세요! 해당 이슈를 이곳에 적으면 pr merge 이후 해당 이슈는 자동으로 close 됩니다. -->

closes: #Issue Number

## 👨🏻‍💻 테스트 체크리스트

<!-- 테스트 사항이 있다면 작성해 주세요! -->

- 이러한 테스트를 했어요!

<br/>

## 📌 변경 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->

<br/>

Commit:
Feat: signal to general react hook. Loader added

Commit:
Feat: added tooltip and toast to buttons in button group

- Added tooltip to each buttons in Social card to guve better user experience
- Added actions to each buttons in Social card. Toast will pop out from the bottom of the screen.
- Fixed css of toast from shadcn
- TODO: Split ButtonGroup file to smaller files.

Commit:
Feat: ButtonGroup tsx splitted in to smaller files.

Hawardshin commented on components/card/cardUsedInSocialPage/buttonGroup/BlockButton.tsx:
```diff
@@ -0,0 +1,71 @@
+  const sendBlockRequest = async () => {
+    console.log("sendBlockRequest's userId: ", userId);
+    // dummy function to test
+    const response = await new Promise((resolve) =>
+      setTimeout(resolve, 1000)
+    ).then(() => false); // change to false to test block request failed
+    if (response) {
+      console.log('block request sent');
+      return true;
+    } else {
+      console.log('block request failed');
+      // TODO: Add actions to manage error.
+      return false;
+    }
+  };
```
지금 이거 API 코드로 하면 지금 안되나요? 가능하면 API로 바꾸면 좋을듯..!

Honeybeei replied:
아직 그거 어떻게 하는지 몰라서 ㅠㅠ

Hawardshin replied:
저도 내일 알려드릴께용

Hawardshin commented on components/card/cardUsedInSocialPage/buttonGroup/BlockButton.tsx:
```diff
@@ -0,0 +1,71 @@
+    }
+  };
+
+  const {toast} = useToast();
```
저 toast 쓰는 법좀 알려주세요..! 
지금 다 alert로 처리중,,

Honeybeei replied:
ㅇㅎ ㅇㅋㅇㅋ 낼 알려드리겠움

Hawardshin commented on components/loader/SpinningLoader.tsx:
```diff
@@ -0,0 +1,41 @@
+
+export default function SpinningLoader() {
+  return (
+    <div className='absolute inset-0 flex items-center justify-center'>
+      <motion.div
+        className='w-32 h-32 sm:w-52 sm:h-52 bg-custom3' // Tailwind classes for width, height, background
+        animate={{
+          scale: [1, 1.2, 1.2, 1, 1],
+          rotate: [0, 0, 270, 270, 0],
+          borderRadius: ['20%', '20%', '50%', '50%', '20%']
+        }}
+        transition={{
+          duration: 2,
+          ease: 'easeIn',
+          times: [0, 0.2, 0.5, 0.8, 1],
+          repeat: Infinity,
+          repeatDelay: 1
+        }}
+      />
+      {/* Loading text animation */}
+      <motion.p
+        className='absolute text-3xl sm:text-5xl font-bold text-center text-custom4'
+        animate={{
+          opacity: [0, 1, 1, 0, 0],
+          scale: [1, 1, 1.2, 1.2, 1],
+          rotate: [0, 0, 270, 270, 0]
+        }}
+        transition={{
+          duration: 2,
+          ease: 'easeInOut',
+          times: [0, 0.2, 0.5, 0.8, 1],
+          repeat: Infinity,
+          repeatDelay: 1
+        }}
+      >
+        Loading
+      </motion.p>
+    </div>
+  );
+}
```
고마워요 안그래도 필요했는데 감사합니다!

Honeybeei replied:
조만간 새로운 로더도 심심할 때 만들 예정

Hawardshin reviewed (approved):
LGTM 잘 쓰겠습니다 로딩창 땡큐 쏘머치

daejlee reviewed (approved):
엘쥐티엠~~
로딩바 개쩐다

Commit:
Feat: removed tooltip, added modal component

- removed tooltip. shadcn sucks. It is a error generator
- shadcn sucks. Therefore, I made my own modal.
